lib/publiccloud/provider.pm

```diff
@@ -309,8 +349,16 @@ sub terraform_apply {
     $self->terraform_applied(1);
 
     my $output = decode_json(script_output("terraform output -json"));
-    my $vms    = $output->{vm_name}->{value};
-    my $ips    = $output->{public_ip}->{value};
+    my $vms;
+    my $ips;
+    if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
+        $vms = $output->{cluster_nodes_names}->{value};
+        $ips = $output->{cluster_nodes_ip}->{value};
```

Comment:
When `PUBLIC_CLOUD_SLES4SAP` is not set you can do this as the terraform output has something like this:

```
public_ip = [
  "54.93.253.246",
]
```

Which is an array with only one element.

In the case of `ha-sap-terraform-deployments` you get an array that contains an array:

```
cluster_nodes_ip = [
  [
    "54.93.142.84",
    "3.121.42.27",
  ],
]
```

As seen on: http://1b210.qa.suse.de/tests/5957#step/sles4sap/97

So most likely you need to do `$vms = $output->{cluster_nodes_ips}->{value}->[0];`, or better yet fix the `outputs.tf` file in the other repository so the array contains the IPs and not an array of an array of the IPs.

I think this change may have been introduced with the changes to terraform 12 in ha-sap-terraform-deployments. Previously, the `output.tf` files had something like this:

```
output "cluster_nodes_ip" {
  value = ["${data.azurerm_public_ip.clusternodes.*.ip_address}"]
}
```

And this expanded to an array of IPs like this:

```
cluster_nodes_ip = [
    23.97.152.184,
    23.97.157.50
]
```

(Tested this yesterday myself with terraform 11)

Now, the outputs.tf files have:

1. AWS: 
```
output "cluster_nodes_ip" {
  value = [aws_instance.clusternodes.*.public_ip]
}
```
2. Azure:
```
output "cluster_nodes_ip" {
  value = [data.azurerm_public_ip.clusternodes.*.ip_address]
}
```
3. GPC:
```
output "cluster_nodes_ip" {
  value = google_compute_instance.clusternodes.*.network_interface.0.access_config.0.nat_ip
}
```

I could see these expanding to an array of arrays in Azure and AWS but not in GCP.

Reply:
I modified the output to ensure that we now don't have double array.

Thx for the hint!